.github/workflows/main.yml

```diff
@@ -0,0 +1,85 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+name: PR Check CI
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  check:
+    runs-on: ubuntu-18.04
+    container: liuguo09/ubuntu-nuttx:18.04
+    
+    steps:
+    - name: Checkout nuttx repo
+      uses: actions/checkout@v2
+      with:
+        path: nuttx
+        fetch-depth: 0
+
+    - name: Checkout apps repo
+      uses: actions/checkout@v2
+      with:
+        repository: apache/incubator-nuttx-apps
+        path: apps
+        fetch-depth: 0
+
+    - name: Check Pull Request
+      run: |
+        cd nuttx
+        git log --oneline -5
+        ranges=`git log -1 --merges --pretty=format:%P | awk -F" " '{ print $1 ".." $2 }'`
+        commits=`git log --reverse --format=format:%H $ranges`
+        echo "./tools/checkpatch.sh -g $commits"
+        ./tools/checkpatch.sh -g $commits
+        
+  build:
+    needs: check
+    runs-on: ubuntu-18.04
+    container: liuguo09/ubuntu-nuttx:18.04
+    
+    strategy:
+      matrix:
+        boards: [sim, mips-riscv-x86, arm-01, arm-02, arm-03, arm-04, arm-05, arm-06, arm-07, arm-08, arm-09, arm-10, arm-11, arm-12, arm-13, arm-14, arm-15]
+
+    steps:
+    - name: Checkout nuttx repo
+      uses: actions/checkout@v2
+      with:
+        path: nuttx
+        fetch-depth: 0
+
+    - name: Fetch nuttx tags
+      run: |
+        cd nuttx
+        git fetch --tags
```

Comment:
Why need fetch tag?

Reply:
Since tags is needed to generate the .version file, or it would build break at first : (

Reply:
Ok

Reply:
This can be done in the checkout step https://github.com/actions/checkout#fetch-all-tags

Reply:
> This can be done in the checkout step https://github.com/actions/checkout#fetch-all-tags

Since I'm using the _with_ option for checkout@v2 action, the _run_ step seems not work together. I'll double check it.

Reply:
I have checked that the instructions in the https://github.com/actions/checkout#fetch-all-tags run two steps too.
So it doesn't make much sense to use same as it.

Reply:
I guess the cost is low for the two steps, but other examples there like submodules show two run lines which I think makes sense to just get everything done. But these are really later optimizations.